contracts/src/SimplePDPService.sol

```diff
@@ -170,9 +164,14 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
             uint256 periodsSkipped = 1;
             if (provingDeadlines[proofSetId] < block.number) {
                 periodsSkipped = 1 + ((block.number - provingDeadlines[proofSetId]) / getMaxProvingPeriod());
+                provingDeadlines[proofSetId] = block.number + getMaxProvingPeriod(); // reset deadline
```

Comment:
Shouldn't that also be `provingDeadlines[proofSetId] = provingDeadlines[proofSetId] + getMaxProvingPeriod();` to keep the deadline in a fixed window?

The current way the proving window will keep sliding backwards as I have to compute and submit the proof some time before the deadline

Reply:
SGTM but we need to catch up multiple windows if we missed multiple otherwise you need to call this a bunch.  So you'll need to add periodsSkipped I think

Reply:
Can't PR now but fee lfree to push

Reply:
Some [discussion about policy options](https://github.com/FILCAT/pdp/pull/63#discussion_r1807935410) here, it wants a bit of thought.